plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRESTCatalog.java

```diff
@@ -0,0 +1,364 @@
+    public Optional<ConnectorMaterializedViewDefinition> getMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void renameMaterializedView(ConnectorSession session, SchemaTableName source, SchemaTableName target)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support materialized views");
+    }
+
+    @Override
+    public void updateColumnComment(ConnectorSession session, SchemaTableName schemaTableName, ColumnIdentity columnIdentity, Optional<String> comment)
+    {
+        loadTable(session, schemaTableName).updateSchema().updateColumnDoc(columnIdentity.getName(), comment.orElse(null)).commit();
+    }
+
+    @Override
+    public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void updateViewComment(ConnectorSession session, SchemaTableName schemaViewName, Optional<String> comment)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support updating view comments");
+    }
+
+    @SuppressModernizer
+    private SessionCatalog.SessionContext convert(ConnectorSession session)
+    {
+        String sessionId = format("%s-%s", session.getUser(), session.getSource().orElse("default"));
+        String identity;
+
+        if (session.getIdentity().getPrincipal().isPresent()) {
+            identity = session.getIdentity().getPrincipal().get().getName();
+        }
+        else {
+            identity = session.getUser();
+        }
+
+        Map<String, String> properties = Map.of(
+                "identity", identity,
+                "user", session.getUser(),
+                "source", session.getSource().orElse(""),
+                "trinoCatalog", catalogName.toString(),
+                "trinoVersion", trinoVersion);
+
+        Map<String, Object> extraCredentialsClaims = session.getIdentity().getExtraCredentials()
+                .entrySet()
+                .stream()
+                .collect(Collectors.toMap(
+                        e -> "extra:" + e.getKey(),
+                        Map.Entry::getValue));
+
+        Map<String, Object> claims = ImmutableMap.<String, Object>builder()
+                .putAll(properties)
+                .putAll(extraCredentialsClaims)
+                .build();
+
+        String subjectJwt = new DefaultJwtBuilder()
+                .setSubject(identity)
+                .setIssuer(trinoVersion)
+                .setIssuedAt(new Date())
+                .addClaims(claims)
+                .serializeToJsonWith(new JacksonSerializer<>())
+                .compact();
+
+        Map<String, String> credentials = ImmutableMap.<String, String>builder()
+                .putAll(session.getIdentity().getExtraCredentials())
+                .put(OAuth2Properties.JWT_TOKEN_TYPE, subjectJwt)
+                .build();
```

Comment:
Why not `buildOrThrow` here and `claims`? Thus, `@SuppressModernizer` is no longer needed.